src/login.service.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  BadRequestException,
+  Injectable,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { User } from './entities/user.entity';
+import { JWT, JWTResult } from './typings/jwt.typing';
+
+@Injectable()
+export class LoginService {
+  constructor(
+    @InjectRepository(User)
+    private usersRepository: Repository<User>,
+    private jwtService: JwtService,
+  ) {}
+
+  async login(username: string, password: string): Promise<JWTResult> {
+    const emptyCredentials = !username || !password;
+
+    if (emptyCredentials) {
+      throw new BadRequestException();
+    }
+
+    const user = await this.usersRepository.findOne({ email: username });
+
+    if (!user) {
+      throw new UnauthorizedException();
+    }
+
+    const isInvalidPassword = !user.isValidPassword(password);
+
+    if (isInvalidPassword) {
+      throw new UnauthorizedException();
+    }
+
+    const payload: JWT = { sub: user.id };
+    const jwt: JWTResult = {
+      accessToken: this.jwtService.sign(payload),
+    };
```

Comment:
I suggest renaming this variables:
```suggestion
    const jwt: JWT = { sub: user.id };
    const jwtResult: JWTResult = {
      accessToken: this.jwtService.sign(payload),
    };
```

Reply:
Resolved on 0ca3d0f8dfebe6ff796a0136a1d1167b753f941a.